### DISCUSSION TITLE
Having a "name" option instead of displaying the whole action

### DISCUSSION THREAD

**Discussion by rooselle** [captainhook-git/captainhook#258] [Ideas]
Instead of displaying the whole command (written in the `action` key of each elements of the `actions` of a hook), it would be nice to be able to set a `name` option that would override this in the CLI output. So instead of this:
![image](https://github.com/user-attachments/assets/f6d10e0d-44e1-4ded-bce5-3034f1cf1c41)
We could have this:
```
 - lint                    : skipped
 - phpstan                 : skipped
 - php-cs-fixer            : skipped
 ```
If actions were configured like this in `captainhook.json`:
```json
{
    "action": "docker compose exec -u sail -T app_laravel npm run lint {$STAGED_FILES|of-type:js}",
    "name": "lint"
},
{
    "action": "docker compose exec -u sail -T app_laravel vendor/bin/phpstan analyse {$STAGED_FILES|of-type:php}",
    "name": "phpstan"
},
{
    "action": "docker compose exec -u sail -T app_laravel vendor/bin/php-cs-fixer fix {$STAGED_FILES|of-type:php}",
    "name": "php-cs-fixer"
}
```

What do you think?

**Comment by Chemaclass** [comment]
Nice idea

**Comment by rooselle** [comment]
Oops actually I'm told it already exists by adding this option:
```
{
    "action": "docker compose exec -u sail -T app_laravel npm run lint {$STAGED_FILES|of-type:js}",
    "config": {
        "label": "lint"
    }
},
```
So maybe it would be just nice to have this in the docs!

Edit. I would actually be willing to push a PR to update it but I don't quite know where the documentation is located?

**Comment by sebastianfeldmann** [comment]
The docs can be found in the `gh-pages` branch of the repository. You can open a pull request for the branch if you want.
If not I will put it on the todo-list ;)

**Reply by rooselle** [reply]
Thanks! Just did that :)
